api/mesh/v1alpha1/health_check.proto

```diff
@@ -39,6 +40,58 @@ message HealthCheck {
 
     // Number of consecutive healthy checks before considering a host healthy.
     uint32 healthy_threshold = 4 [ (validate.rules).uint32 = {gt : 0} ];
+
+    // Tcp defines optional configuration for specifying bytes to send and
+    // expected response during the health check
+    message Tcp {
+      // Bytes which will be send during the health check to the target
+      google.protobuf.BytesValue send = 1;
+
+      // Bytes blocks expected as a response. When checking the response,
+      // “fuzzy” matching is performed such that each block must be found, and
+      // in the order specified, but not necessarily contiguous.
+      repeated google.protobuf.BytesValue receive = 2;
+    }
+
+    Tcp tcp = 5;
+
+    // Http defines optional Http configuration which will instruct the service
+    // the health check will be made for is an http service. It's mutually
+    // exclusive with the Tcp block so when provided you can't provide
+    // the Tcp configuration
+    message Http {
+      // The HTTP path which will be requested during the health check
+      // (ie. /health)
+      google.protobuf.StringValue path = 1; // (REQUIRED)
```

Comment:
do we need to differentiate empty string from null? If not, regular string is enough. Same with other wrappers below

Reply:
We actually don't have to, I just did it initially as I was not sure if it would work with providing empty string `""` as a value, then I tested it and it was invalid. I decided to leave it to have more direct error messages during validation but I'm happy to change it as I'm not attached to it :-)